src/rgw/rgw_rest_s3.cc

```diff
@@ -3890,8 +3890,23 @@ AWSGeneralAbstractor::get_auth_data_v4(const req_state* const s,
     throw -EPERM;
   }
 
-  /* Get the expected hash. */
-  auto exp_payload_hash = rgw::auth::s3::get_v4_exp_payload_hash(s->info);
+  bool is_non_s3_op = false;
+  if (s->op_type == RGW_STS_GET_SESSION_TOKEN || s->op_type == RGW_STS_ASSUME_ROLE) {
+    is_non_s3_op = true;
+  }
+
+  const char* exp_payload_hash = nullptr;
+  string payload_hash;
+  if (is_non_s3_op) {
+    //For non s3 ops, we need to calculate the payload hash
+    //const auto hash = rgw::auth::s3::calc_v4_payload_hash(s->info);
```

Comment:
is there a future direction for the commented case?  please elaborate

Reply:
The commented part has to be removed.